### PR TITLE
Signup to Homepage flow

### PR DESCRIPTION
**Pull request by t-schell**

### CODE REVIEW

**Comment by t-schell** [comment]
This PR is dependent on back-end counterpart

File: src/components/LoginSignup/LoginSignup.js [hunk 1]
```diff
@@ -112,65 +154,174 @@ export default class LoginSignup extends React.Component {
...
+        props.openOrCloseLoginModal();
+        window.location.href = `/Home?user=${res.data.userId}`
+      } else if(res.status === 401) {
+        this.setState({ serviceErrMsg: "Invalid username or password"})
+      } else {
+        this.setState({ serviceErrMsg: "Something went wrong."})
+      }
+    }).catch(error => {
+      this.setState({ serviceErrMsg: SERVICE_ERROR_MESSAGE })
+      console.error('Login post call failed: ' + error.message);
```
Comment (mchuong1) [inline src/components/LoginSignup/LoginSignup.js:227]: I believe the response message should come from the backend or the backend should also have the error message as well.
All the front end should do is relay the message
something like
`else{
 this.setState({serviceErrMsg: res.message})
}`

File: src/components/LoginSignup/LoginSignup.js [hunk 2]
```diff
@@ -112,65 +154,174 @@ export default class LoginSignup extends React.Component {
...
+    var props = this.props
+    var email = this.state.email
+    var password = await encrypt(this.state.password)
+
+
+    // post login
+    dataSource.PostData("login", { email : email, password : password})
+    .then(function(res) {
+      if(res.status === 200) {
+        props.openOrCloseLoginModal();
```
Comment (mchuong1) [inline src/components/LoginSignup/LoginSignup.js:218]: You can use destructuring here and should be able to call the function without the props leading in front
`const { openOrCloseLoginModal  } = this.props;`
`openOrCloseLoginModal();`

Comment (mchuong1) [reply src/components/LoginSignup/LoginSignup.js:218]: or actually have the openOrCloseModal() have a bigger scope so we dont have to declare var props = this.props multiple times

File: src/components/LoginSignup/LoginSignup.js [hunk 3]
```diff
@@ -112,65 +154,174 @@ export default class LoginSignup extends React.Component {
...
+  validateLogin() {
+    var validEmail = this.validateEmail();
+    var validPwd = this.validatePwd();
+    return validEmail && validPwd;
+  }
+
+  async executeLogin() {
+    var props = this.props
+    var email = this.state.email
+    var password = await encrypt(this.state.password)
```
Comment (mchuong1) [inline src/components/LoginSignup/LoginSignup.js:211]: this could be renamed to encryptedPassword for clarity

**Review by mchuong1 — APPROVED** [review]

**Pull request merged**